diff --git a/drivers/leds/Kconfig b/drivers/leds/Kconfig
index 56e8198e1..76ee53271 100644
--- a/drivers/leds/Kconfig
+++ b/drivers/leds/Kconfig
@@ -371,6 +371,15 @@ config LEDS_GPIO
 	  defined as platform devices and/or OpenFirmware platform devices.
 	  The code to use these bindings can be selected below.
 
+config LEDS_LP3943
+        tristate "LED Support for N.S. LP3943 (Fun Light) I2C chip"
+        depends on LEDS_CLASS
+        depends on I2C
+        help
+          This option enables support for LEDs connected to the National
+          Semiconductor LP3943 Lighting Management Unit (LMU) also known as
+          Fun Light Chip.
+
 config LEDS_LP3944
 	tristate "LED Support for N.S. LP3944 (Fun Light) I2C chip"
 	depends on LEDS_CLASS
